internal/request/request.go

```diff
@@ -17,27 +15,41 @@ import (
 	"github.com/RedHatInsights/catalog_mqtt_client/internal/logger"
 	"github.com/RedHatInsights/catalog_mqtt_client/internal/tarwriter"
 	"github.com/RedHatInsights/catalog_mqtt_client/internal/towerapiworker"
-	mqtt "github.com/eclipse/paho.mqtt.golang"
 	log "github.com/sirupsen/logrus"
 )
 
 // Handler interface allows for easy mocking during testing
 type Handler interface {
-	StartHandlingRequests(mqttClient mqtt.Client, config *common.CatalogConfig, wh towerapiworker.WorkHandler)
+	StartHandlingRequests(config *common.CatalogConfig, wh towerapiworker.WorkHandler)
 	//parseRequest(b []byte) (*RequestMessage, error)
 }
 
 // DefaultRequestHandler implements the 3 RequestHandler methods
 type DefaultRequestHandler struct {
 }
 
+type listener interface {
+	stop()
+}
+
 // StartHandlingRequests starts a MQTT listener. It will not stop until receives a system signal.
-func (drh *DefaultRequestHandler) StartHandlingRequests(mqttClient mqtt.Client, config *common.CatalogConfig, wh towerapiworker.WorkHandler) {
-	defer mqttClient.Disconnect(10)
+func (drh *DefaultRequestHandler) StartHandlingRequests(config *common.CatalogConfig, wh towerapiworker.WorkHandler) {
 	sigs := make(chan os.Signal, 1)
 	shutdown := make(chan struct{})
 	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
-	startMQTTListener(mqttClient, config, wh, shutdown)
+
+	if config.GRPCPort != 0 {
```

Comment:
@bzwei This is most probably coming in the enviornment variable, currently it is a static unix domain socket

https://github.com/RedHatInsights/yggdrasil/blob/a1b89b41192eb3b5cdb71f312b474ce3cb04a662/dispatcher.go#L139

Reply:
OK. We can change it later when we know for sure what the configuration will be. With this current setting we can start local testing.